test/Microsoft.DotNet.ShellShimMaker.Tests/ShellShimMakerTests.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.DotNet.ShellShimMaker.Tests
+{
+    public class ShellShimMakerTests : TestBase
+    {
+        [Fact]
+        public void GivenAnExecutablePathItCanGenerateShimFile()
+        {
+            var outputDll = MakeHelloWorldExecutableDll();
+
+            var muxer = new Muxer();
+            var shellShimMaker = new ShellShimMaker(Path.GetDirectoryName(muxer.MuxerPath));
```

Comment:
Why are we putting these shims in the dotnet folder? Seems very wrong to me. If something happens and the test does not clean up, we will have this trash in the actual product.

Reply:
fixed, now it is a temp folder